src/io/SPI.cpp

```diff
@@ -1,5 +1,13 @@
 #include <EVT/io/SPI.hpp>
 
+#define CONCAT(a, b) CONCAT_INNER(a, b)
+#define CONCAT_INNER(a, b) a##b
+
+#define SPI_CHECK(functionCall, statusID)              \
```

Comment:
I did some testing and came up with this solution:
```suggestion
#define SPI_CHECK(functionCall)                        \
    SPIStatus CONCAT(status, __LINE__) = functionCall; \
    if (CONCAT(status, __LINE__) != SPIStatus::OK)     \
    return CONCAT(status, __LINE__)
```

This would allow you to eliminate the `statusID`, making the macro completely self-contained. From what I've seen, I think this is the ideal solution.